### PR TITLE
`ultralytics 8.3.54` New Streamlit inference Solution

### PR DESCRIPTION
**Pull request by RizwanMunawar**
@glenn-jocher Hi, this PR converts the streamlit application to a proper solution to maintain consistency, make it easily manageable, and provide a better user experience. Thank you!

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR introduces a revamped user experience for **Streamlit-based real-time inference** and minor textual refinements across the Ultralytics repository while updating the package version to `8.3.54`. 🚀

### 📊 Key Changes
- ✨ **Streamlit Inference Overhaul**: Reworked Streamlit apps with a new `Inference` class, enabling a more user-friendly UI and functionality for real-time predictions, object tracking, and class selection.
  - Added sidebar configurations for video source, model selection (including custom models), and detection settings (e.g., confidence threshold).
  - New, interactive live inference dashboard supports webcam and video uploads.
  - Introduced annotation and FPS display during real-time inference.
- 🎨 **Refinements**:
  - Transitioned from `solutions.inference()` to `solutions.Inference().inference()` for better modularity.
  - Enhanced documentation for new Streamlit functionality.
- 🪄 **Command-line Updates**: Replaced outdated `streamlit-predict` CLI command with `yolo solutions inference` for launching Streamlit apps with custom models.
- 🔄 Minor textual consistency improvements in READMEs, guides, and code comments (e.g., standardized "Ultralytics" capitalization).
- 🛠️ Small code formatting tweaks across examples and utility scripts for improved readability.

### 🎯 Purpose & Impact
- **Enhanced User Experience**: 
  - Easier setup and usage of the Streamlit inference feature, appealing to both beginners and advanced users. 🎥
  - Users can load models, configure detection thresholds, and visualize live predictions seamlessly.
- **Improved Maintainability**:
  - Modular `Inference` class structure simplifies future feature extensions.
  - Standardized naming and cleaned-up code ensure better readability for contributors. 🙌
- **Backward Compatibility**:
  - Familiar CLI commands transitioned smoothly to intuitive replacements.
  - Existing workflows can incorporate custom models effortlessly. 💡

This update makes **real-time detection accessible and visually engaging**, strengthening Ultralytics' framework for rapid prototyping and demonstrations. 🎉

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any relevant issues. Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics Continuous Integration (CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant documentation for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our Contributor License Agreement (CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please review the project’s contributing guidelines or leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀 An Ultralytics engineer will review your submission soon. 😊

File: ultralytics/solutions/streamlit_inference.py [hunk 1]
```diff
@@ -4,145 +4,186 @@
 import time
 
 import cv2
-import torch
 
-from ultralytics.utils.checks import check_requirements
+from ultralytics.solutions.solutions import LOGGER, YOLO, check_requirements
```
Comment (glenn-jocher) [inline ultralytics/solutions/streamlit_inference.py:8]: @RizwanMunawar these should be imported from their root package location, not other locations, so i.e. `from ultralytics import YOLO`.

Comment (RizwanMunawar) [reply ultralytics/solutions/streamlit_inference.py:8]: @glenn-jocher Hi, good catch. I am going to update it now! Thanks

**Comment by RizwanMunawar** [comment]
@glenn-jocher Great idea! Fixed for all solutions. Thank you!

**Comment by UltralyticsAssistant** [comment]
🎉 This fantastic PR is now merged — a huge thank you to @RizwanMunawar and @glenn-jocher for your incredible contributions! 🚀 Your efforts in revamping the Streamlit inference experience and refining Ultralytics' tools are nothing short of game-changing. You've made real-time detection more accessible, intuitive, and engaging for users worldwide. 🌟

As Marcus Aurelius once said, *"What we do now echoes in eternity."* Your work will undoubtedly inspire and empower countless users and developers to create, innovate, and push the boundaries of what's possible. 💡💪

Keep up the amazing work, and thank you for making the Ultralytics community stronger! 🙌

**Pull request merged**